server/stream.go

```diff
@@ -1013,25 +1009,12 @@ func (mset *Stream) processInboundJetStreamMsg(_ *subscription, pc *client, subj
 		for _, o := range obs {
 			o.incStreamPending(seq, subject)
 			if !o.deliverCurrentMsg(subject, hdr, msg, seq, ts) {
-				needSignal = true
+				o.signalNewMessages()
```

Comment:
Should we have a locked/unlocked version?

Reply:
I don't think so since these are channels. Just can't nil them out ever which is fine.

Reply:
Read from multiple Go routines ok.